frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/SlideInBackgroundable.java

```diff
@@ -0,0 +1,158 @@
+package com.virtuslab.gitmachete.frontend.actions.backgroundables;
+
+import static com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle.getString;
+import static com.virtuslab.gitmachete.frontend.vfsutils.GitVfsUtils.getMacheteFilePath;
+import static java.text.MessageFormat.format;
+
+import java.nio.file.Path;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import com.intellij.openapi.progress.ProgressIndicator;
+import com.intellij.openapi.progress.Task;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vcs.VcsNotifier;
+import git4idea.GitLocalBranch;
+import git4idea.repo.GitRepository;
+import io.vavr.collection.List;
+import io.vavr.control.Try;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import com.virtuslab.branchlayout.api.BranchLayoutEntry;
+import com.virtuslab.branchlayout.api.EntryDoesNotExistException;
+import com.virtuslab.branchlayout.api.EntryIsDescendantOfException;
+import com.virtuslab.branchlayout.api.EntryIsRootException;
+import com.virtuslab.branchlayout.api.IBranchLayout;
+import com.virtuslab.branchlayout.api.IBranchLayoutEntry;
+import com.virtuslab.branchlayout.api.readwrite.IBranchLayoutWriter;
+import com.virtuslab.gitmachete.frontend.actions.dialogs.SlideInOptions;
+
+public class SlideInBackgroundable extends Task.Backgroundable {
+
+  private final GitRepository gitRepository;
+  private final IBranchLayout branchLayout;
+  private final IBranchLayoutWriter branchLayoutWriter;
+  private final Runnable preSlideInRunnable;
+  private final SlideInOptions slideInOptions;
+  private final String parentName;
+  private final VcsNotifier notifier;
+
+  public SlideInBackgroundable(
+      Project project,
+      GitRepository gitRepository,
+      IBranchLayout branchLayout,
+      IBranchLayoutWriter branchLayoutWriter,
+      Runnable preSlideInRunnable,
+      SlideInOptions slideInOptions,
+      String parentName) {
+    super(project, getString("action.GitMachete.BaseSlideInBranchBelowAction.task-title"));
+    this.gitRepository = gitRepository;
+    this.branchLayout = branchLayout;
+    this.branchLayoutWriter = branchLayoutWriter;
+    this.preSlideInRunnable = preSlideInRunnable;
+    this.slideInOptions = slideInOptions;
+    this.parentName = parentName;
+    this.notifier = VcsNotifier.getInstance(project);
+  }
+
+  @Override
+  public void run(ProgressIndicator indicator) {
+    preSlideInRunnable.run();
+
+    // `preSlideInRunnable` may perform some sneakily-asynchronous operations (e.g. checkoutRemoteBranch).
+    // The high-level method used within the runnable do not allow us to schedule the tasks after them.
+    // (Stepping deeper is not an option since we loose some important logic or make us very dependent on git4idea.)
```

Comment:
```suggestion
    // (Stepping deeper is not an option since we would lose some important logic or become very dependent on the internals of git4idea.)
```